addons/pandora/model/type.gd

```diff
@@ -0,0 +1,56 @@
+## This class allows us to make it easier 
+## to specify new types, e.g. by adding a new file 
+## which then internally specifies everything Pandora needs to know,
+## including serialization, property settings and validation.
+class_name PandoraPropertyType extends RefCounted
+
+
+class UndefinedType extends PandoraPropertyType:
```

Comment:
This type is useful to avoid `null` checks but have instead an `undefined` type.